### ISSUE TITLE
distortos::DynamicThread' has no member named 'detach'

### ISSUE THREAD

**Issue by adamvm** [DISTORTEC/distortosExamples#2]
cmake .. -DCMAKE_TOOLCHAIN_FILE=../distortos/source/board/ST_STM32F4DISCOVERY/Toolchain-ST_STM32F4DISCOVERY.cmake -GNinja
-- The C compiler identification is GNU 7.3.1
-- The CXX compiler identification is GNU 7.3.1
-- Check for working C compiler: /usr/bin/arm-none-eabi-gcc
-- Check for working C compiler: /usr/bin/arm-none-eabi-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/arm-none-eabi-g++
-- Check for working CXX compiler: /usr/bin/arm-none-eabi-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Generating /home/adi/dev/distortosExamples/output/distortos/distortosConfiguration.cmake
-- Generating /home/adi/dev/distortosExamples/output/distortos/include/distortos/distortosConfiguration.h
-- CMAKE_BUILD_TYPE not set, defaulting to RelWithDebInfo
-- Auto-enabling distortos_Peripherals_GPIOA. Dependents: buttons B1
-- Auto-enabling distortos_Peripherals_GPIOD. Dependents: leds Ld3, leds Ld4, leds Ld5, leds Ld6
-- Generating /home/adi/dev/distortosExamples/output/distortos/ST_STM32F4DISCOVERY.preprocessed.ld
-- Generating /home/adi/dev/distortosExamples/output/.gitignore
-- Configuring done
-- Generating done
-- Build files have been written to: /home/adi/dev/distortosExamples/output
adi@W540:~/dev/distortosExamples/output$ cmake-gui ..
adi@W540:~/dev/distortosExamples/output$ ninja
[108/113] Building CXX object threadExamples/dynamicThreadBlinker/CMakeFiles/dynamicThreadBlinker.dir/dynamicThreadBlinker.cpp.o
FAILED: threadExamples/dynamicThreadBlinker/CMakeFiles/dynamicThreadBlinker.dir/dynamicThreadBlinker.cpp.o 
/usr/bin/arm-none-eabi-g++  -DLFS1_NO_DEBUG -DLFS1_NO_ERROR -DLFS1_NO_MALLOC -DLFS1_NO_WARN -Idistortos/include -I../distortos/include -I../distortos/source/board/ST_STM32F4DISCOVERY/include -I../distortos/source/chip/STM32/peripherals/GPIOv2/include -I../distortos/source/chip/STM32/peripherals/SPIv1/include -I../distortos/source/chip/STM32/peripherals/USARTv1/include -I../distortos/source/chip/STM32/peripherals/DMAv2/include -I../distortos/source/chip/STM32/STM32F4/../include -I../distortos/source/chip/STM32/STM32F4/include -I../distortos/source/chip/STM32/STM32F4/external/CMSIS-STM32F4 -I../distortos/source/architecture/ARM/ARMv6-M-ARMv7-M/include -I../distortos/source/architecture/ARM/ARMv6-M-ARMv7-M/external/CMSIS -I../distortos/source/FileSystem/littlefs1/external/littlefs1 -fno-rtti -fno-exceptions -ffunction-sections -fdata-sections -Wall -Wextra -Wshadow -Wno-psabi -mcpu=cortex-m4 -mfpu=fpv4-sp-d16 -mthumb -mfloat-abi=hard -fno-use-cxa-atexit -O2 -g -ggdb3   -fdiagnostics-color=always -std=gnu++11 -MD -MT threadExamples/dynamicThreadBlinker/CMakeFiles/dynamicThreadBlinker.dir/dynamicThreadBlinker.cpp.o -MF threadExamples/dynamicThreadBlinker/CMakeFiles/dynamicThreadBlinker.dir/dynamicThreadBlinker.cpp.o.d -o threadExamples/dynamicThreadBlinker/CMakeFiles/dynamicThreadBlinker.dir/dynamicThreadBlinker.cpp.o -c ../threadExamples/dynamicThreadBlinker/dynamicThreadBlinker.cpp
../threadExamples/dynamicThreadBlinker/dynamicThreadBlinker.cpp: In function 'int main()':
../threadExamples/dynamicThreadBlinker/dynamicThreadBlinker.cpp:107:36: error: 'class distortos::DynamicThread' has no member named 'detach'
    std::chrono::milliseconds{401}).detach();
                                    ^~~~~~
../threadExamples/dynamicThreadBlinker/dynamicThreadBlinker.cpp:125:17: error: 'class distortos::DynamicThread' has no member named 'detach'
     periods[i]).detach();
                 ^~~~~~
[110/113] Building CXX object threadExamples/staticThreadBlinker/CMakeFiles/staticThreadBlinker.dir/staticThreadBlinker.cpp.o
ninja: build stopped: subcommand failed.

**Comment by FreddieChopin** [comment]
Hello Adam!

By default thread detachment is disabled. So if you create a new configuration (as you did with `cmake .. -DCMAKE_TOOLCHAIN_FILE=../distortos/source/board/ST_STM32F4DISCOVERY/Toolchain-ST_STM32F4DISCOVERY.cmake -GNinja`) it has this feature disabled. When you build the application that needs it - like the dynamic thread blinker - it will fail. However you can simply enable it in the GUI with `cmake-gui ..` - look for `distortos_Scheduler_03_Support_for_thread_detachment`.

Please note that even when you do that the application you'll build that way may still crash. This is because sometimes a bigger-than-default stack is needed, especially in the case of static blinker application (this one is going to crash for sure with default stack size if you have more LEDs and more blinker threads). This is why I would recommend to start with the configurations included in the repository. This way you can start with something that is known to work and modify it to suit your needs - or just see what is changed between there and the default (this can be easily done by comparing `distortosConfiguration.cmake` files with `diff -u ...`).

For this board I would therefore recommend to start with:
`cmake -C../configurations/ST_STM32F4DISCOVERY/distortosConfiguration.cmake .. -GNinja`
(make sure to COMPLETELY delete the output folder before starting again!)

**Comment by adamvm** [comment]
Works
